core/src/main/scala/kafka/tools/ConsumerPerformance.scala

```diff
@@ -107,11 +123,14 @@ object ConsumerPerformance {
 
   }
 
-  private[tools] def printHeader(showDetailedStats: Boolean): Unit = {
-    if (showDetailedStats)
-      println("start.time, end.time, data.consumed.in.MB, MB.sec, data.consumed.in.nMsg, nMsg.sec")
-    else
-      println("time, threadId, data.consumed.in.MB, MB.sec, data.consumed.in.nMsg, nMsg.sec")
+  private[tools] def printHeader(showDetailedStats: Boolean, useOldConsumer: Boolean): Unit = {
+    val newFieldsInHeader = s"${if (!useOldConsumer) ", total.rebalance.time.ms, fetch.time.ms, fetch.MB.sec, fetch.nMsg.sec" else ""}"
+    if (!showDetailedStats) {
+        println("start.time, end.time, data.consumed.in.MB, MB.sec, data.consumed.in.nMsg, nMsg.sec" + newFieldsInHeader)
+      } else {
+        println("time, threadId, data.consumed.in.MB, MB.sec, data.consumed.in.nMsg, nMsg.sec" +
+          s"${if (useOldConsumer) "" else ", rebalance.time.ms" + newFieldsInHeader}")
```

Comment:
You mean all these lines are redundant? Isn't it printing the header based on whether using old or new consumer and whether printing in detailed?

Reply:
I meant that the check for `useOldConsumer` is redundant give how we built `newFieldsInHeader`.

Reply:
Checking `useOldConsumer` here is because new consumer should show the rebalance time for  each interval as you suggest. However, the old consumer does not get this field displayed. Does it make sense?